cli/cmd/compose/create.go

```diff
@@ -34,10 +35,14 @@ func createCommand(p *projectOptions) *cobra.Command {
 		Use:   "create [SERVICE...]",
 		Short: "Creates containers for a service.",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Build && opts.noBuild {
+				return fmt.Errorf("--build and --no-build are incompatible")
+			}
```

Comment:
Nit: we should also add a check for force-recreate and no-recreate, next to this one. (Check already exists in the up command)